merge/update.go

```diff
@@ -83,6 +83,8 @@ func (s *Updater) update(oldObject, newObject *typed.TypedValue, version fieldpa
 			versions[managerSet.APIVersion()] = compare
 		}
 
+		compare.Remove(ignored)
```

Comment:
compare is coming out of a cache here (`versions`), can we do the removal before we store in the cache?

Reply:
While thinking that "ignored" isn't per request, I realized that it's "per version". which means that this might be incorrect because ignored doesn't work for all versions.

Reply:
But we pass different ignores when creating the fieldManager and they get used here, don't they?
So we could put ignored in the updater as subresources and versions all get their own fieldManager

Reply:
Also I first hat the remove above on line 82, but that didn't work because it got skipped when the version was already present.

Reply:
Right, at that line we do set thing inside the cache, but if you look at it, we're also setting into the cache at the very beginning of the method.